salt/_modules/metalk8s_drain.py

```diff
@@ -507,41 +443,55 @@ def wait_for_eviction(self, pods):
             total_t += time.time() - iteration_start
         return pods
 
-    def evict_pod(self, pod):
-        '''Trigger the eviction process for a single pod.
 
-        Args:
-          - pod: the Kubernetes API pod object to evict
-        Returns: None
-        Raises: CommandExecutionError in case of API error
-        '''
-        delete_options = V1DeleteOptions()
-        if self.grace_period >= 0:
-            delete_options.grace_period = self.grace_period
+def evict_pod(name, namespace='default', grace_period=1,
+              kubeconfig=None, context=None):
+    '''Trigger the eviction process for a single pod.
+
+    Args:
+        - name          : the name of the pod to evict
+        - namespace     : the namespace of the pod to evict
+        - grace_period  : the time to wait before killing the pod
+    Returns: api response
+    Raises: CommandExecutionError in case of API error
+    '''
+    kind_info = __utils__['metalk8s_kubernetes.get_kind_info']({
+        'kind': 'Pod',
+        'apiVersion': 'v1'
+    })
```

Comment:
That's a strange indirect way of getting the API you're looking for...
The stuff in `__utils__` is really meant for use in the generic CRUD methods applied on manifests, maybe if you want something for manipulating any API, you could add something lighter than this `get_kind_info`.
Since we're importing stuff from `kubernetes.client` anyway, why not just import the API object directly? Configuring the API client can then be done using `kubernetes.config`.

Feel free to ignore and postpone in a later debt ticket ;)

Reply:
For me we want `kubernetes_utils.py` to be the only one interacting with `python-kubernetes`. 
Here we only import object to put in the body and exception.

And btw this method is kind of "CRUD method" as we create a "pod eviction"

Reply:
No you're using the wrapper `ObjectInfo` for `v1/Pod`, to get its `ApiClient` wrapper, to then get the `CoreV1Api` instance (through `kind_info.client.api`).

My point is, instead of doing this dance, just grab `CoreV1Api` and be done with it, you don't need wrappers from `__utils__` here. `ApiClient` is useful to have some nice `create`/`delete`/... method names, but the method you use is `create_namespaced_pod_eviction`. So, if you **really** want to use `kubernetes_utils` here, please do something like this:

```python
# in kubernetes_utils
KNOWN_STD_KINDS = {
    [...]
    ('policy/v1beta1', 'Eviction'): KindInfo(
        model=k8s_client.V1beta1Eviction,
        api_cls=k8s_client.CoreV1Api,
        name='namespaced_pod_eviction',
    ),
   [...]
}

# in this file
delete_options = {'grace_period': grace_period} if grace_period >= 0 else {}
manifest = {
    'kind': 'Eviction',
    'apiVersion': 'policy/v1beta1',
    'metadata': {'name': name, 'namespace': namespace},
    'delete_options': delete_options,
}

try:
    result = __salt__['metalk8s_kubernetes.create_object'](
        name=name,
        manifest=manifest,
        saltenv=__env__,
        **kwargs
    )
except CommandExecutionError as exc:
    raise CommandExecutionError(
        'Failed to evict pod "{}" in namespace "{}": {!s}'.format(
            name, namespace, exc
        )
    )
```

Reply:
I will add an entry in `kubernetes_utils.py`

Reply:
Ok, no we can't because we have no method for exemple `read_namespaced_pod_eviction` and to use it in `KindInfo` we need to have all methods

Reply:
It could be nice to define a utility method, to retrieve a client by name (so that we don't interact with `kubernetes.client` directly) like:

```python
__util__['metalk8s_kubernetes.get_api_client'](
    api_name='CoreV1API',
    name='namespaced_pod_eviction',
)
```

However, since the `ApiClient` uses the `CRUD_METHODS` class attribute to know which methods to look for during `__init__`, it won't work to just expose this object out-of-the-box. Maybe add a way to select a subset of those methods in `__init__` args:

```python
__util__['metalk8s_kubernetes.get_api_client'](
    api_name='CoreV1API',
    name='namespaced_pod_eviction',
    method_names=['read'],
)
```